sources/packages/api/src/route-controllers/user/user.controller.ts

```diff
@@ -1,20 +1,49 @@
-import { Controller, Get, Req } from "@nestjs/common";
+import { Controller, Get, Patch, Req } from "@nestjs/common";
 import { Request } from "express";
 import { AuthService, UserService } from "../../services";
+import BaseController from "../BaseController";
 
 @Controller("users")
-export class UserController {
+export class UserController extends BaseController {
   constructor(
     private readonly service: UserService,
     private readonly authService: AuthService,
-  ) {}
+  ) {
+    super();
+  }
 
   @Get("/check-user")
-  checkUser(@Req() request: Request) {
-    const userInfo = this.authService.parseAuthorizationHeader(
-      request.headers.authorization,
-    );
+  async checkUser(@Req() request: Request) {
+    try {
+      const userInfo = this.authService.parseAuthorizationHeader(
+        request.headers.authorization,
+      );
+      const userInSABC = await this.service.getUser(userInfo.userName);
+      if (!userInSABC) {
+        //TODO: Set a 404 in response when no user is found.
+        //response.status(404);
```

Comment:
If it is needed you can check on the student controller how different HTTP errors are returned, like below.
```suggestion
//response.status(404);
throw new HttpException(
        {
          status: HttpStatus.FORBIDDEN,
          message: "Wrong token",
        },
        HttpStatus.FORBIDDEN,
      );
```

Reply:
Done